doc/Makefile

```diff
@@ -10,11 +10,11 @@ BUILDDIR      = build
 
 # Put it first so that "make" without argument is like "make help".
 help:
-	@$(SPHINXBUILD) -M help $(SPHINXOPTS) $(O) "$(SOURCEDIR)" "$(BUILDDIR)"
+	$(SPHINXBUILD) --help
 
 .PHONY: help Makefile
 
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
-	@$(SPHINXBUILD) -M $@ $(SPHINXOPTS) $(O) "$(SOURCEDIR)" "$(BUILDDIR)"
+	$(SPHINXBUILD) -b $@ $(SPHINXOPTS) $(O) "$(SOURCEDIR)" "$(BUILDDIR)"
```

Comment:
I want to use `-M` at least in local builds, because it makes it easy for me to build different kinds of documentation, and also enables `make clean`.

Considering the fact that having travis build the docs is only a sanity check (the real deployment occurs on readthedocs.io), I really want to avoid a lot of workarounds for its sake.